spring-core/src/main/java/org/springframework/core/io/UrlResource.java

```diff
@@ -167,6 +169,12 @@ private URL getCleanedUrl(URL originalUrl, String originalPath) {
 	public InputStream getInputStream() throws IOException {
 		URLConnection con = this.url.openConnection();
 		ResourceUtils.useCachesIfNecessary(con);
+
+		if (this.url.getUserInfo() != null) {
+			String basicAuth = "Basic " + new String(Base64Utils.encode(url.getUserInfo().getBytes()));
```

Comment:
What about using `Base64Utils.encodeToString()`?

Reply:
@dreis2211 
Thank you for your comment.
I fixed it in "005911e"